src/core/config/mac.h

```diff
@@ -489,16 +489,25 @@
 #endif
 
 /**
- * @def OPENTHREAD_CONFIG_CSL_MIN_RECEIVE_ON
+ * @def OPENTHREAD_CONFIG_MIN_RECEIVE_ON_AHEAD
  *
- * The minimum CSL receive window (in microseconds) required to receive an IEEE 802.15.4 frame.
- * - Maximum frame size with preamble: 6*2+127*2 symbols
- * - AIFS: 12 symbols
- * - Maximum ACK size with preamble: 6*2+33*2 symbols
+ * The minimum time (in microseconds) before the MHR start that the radio should be in receive state in order
```

Comment:
comment would also need "CSL" in it - see previous comment.

Description could say "radio should be in receive state and ready to properly receive ... "  with the word "ready" to emphasize that the radio is set before to the state "receive" but then it's not immediately ready yet.

Reply:
And I'm a bit confused by the two similar-looking constants!

We have `OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD` which is the time between telling the radio to go into receive-mode (at a particular scheduled time, or asap - depending on the platform) and when it's ready to receive, worst-case.

I'm assuming that after this time period the radio is for sure ready to receive a frame correctly and the preamble can arrive.

So the time t at which the radio needs to be "scheduled to receive on" is :

`t = cslTimeMHR - 192 - OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD`

in which case we don't use the other constant `OPENTHREAD_CONFIG_MIN_RECEIVE_ON_AHEAD`.

Or it could be this time:

`t = cslTimeMHR - OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD` in which case we also don't use the other constant `OPENTHREAD_CONFIG_MIN_RECEIVE_ON_AHEAD`.

Or it could be this time:

`t = cslTimeMHR - OPENTHREAD_CONFIG_MIN_RECEIVE_ON_AHEAD - OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD` 

in which case we use both constants, however it's not clear from the `OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD` definition that it means this. And `OPENTHREAD_CONFIG_MIN_RECEIVE_ON_AHEAD`  would always be 192 us because that's determined by the standard and not by the particular platform so it shouldn't be in the platform config then.

Reply:
I believe the confusion comes because `OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD` has two usages:

- When `!RadioSupportsReceiveTiming()`, is as you mention, and the constant is added to the "left" of the CSL sample time.
- When `RadioSupportsReceiveTiming()`, the constant is not taken into account for the duration of the receive slot, it's rather a safe margin used to schedule the delayed receive operation on time, taking into account serialization delays and so on.

What I'm fixing with this PR is the second case, for which currently OT code schedules the start of the receive window matching with the expected MHR time, so there is no margin for SHR + PHR.

This could be fixed with the usage of a constant (192 us), but there is room for energy efficiency improvements since it could be that the radio will get the frame even with partial reception of the preamble. So this allows to save a few microseconds of receive time on each CSL period.

Reply:
Thanks, this resolves the confusion. It makes sense to have both constants then.